File: trunk/PolarMath/src/org/macondo/polar/evaluation/hrv/Mo.java
package org.macondo.polar.evaluation.hrv;

import java.util.List;

import org.macondo.polar.data.Training;
import org.macondo.polar.evaluation.Evaluation;
import org.macondo.polar.util.Histogram;

public class Mo implements Evaluation<Double> {
	
	public Double evaluate(Training training) {
		List<Integer> intervals = training.getIntervals();
		
		Histogram h = new Histogram(intervals.size()).init();
		for (Integer interval : intervals) {
            h.addRRInterval(interval);
        }
		return h.getMaxIntervalStart() / (double) 1000;
	}

}

File: trunk/PolarMathWinRT/Evaluation/HRV/Mo.cs
using HrmMath.Data;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using HrmMath.Util;

namespace HrmMath.Evaluation.HRV
{
    internal sealed class Mo : IEvaluation
    {
        public Index Name
        {
            get { return Index.Mo; }
            set { }
        }

        public object Evaluate(SessionData training) 
        {
		    var intervals = training.Intervals;
		
		    //Histogram h = new Histogram(intervals.size()).init();
		    var h = new Histogram().Init();
		    foreach (var interval in intervals) {
                h.AddRrInterval(interval);
            }
		    return h.GetMaxIntervalStart() / (double) 1000;
	    }
    }
}
